src/main/java/uk/gov/companieshouse/api/accounts/controller/ProfitLossController.java

```diff
@@ -0,0 +1,131 @@
+package uk.gov.companieshouse.api.accounts.controller;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.validation.BindingResult;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import uk.gov.companieshouse.api.accounts.AttributeName;
+import uk.gov.companieshouse.api.accounts.exception.DataException;
+import uk.gov.companieshouse.api.accounts.links.SmallFullLinkType;
+import uk.gov.companieshouse.api.accounts.model.rest.SmallFull;
+import uk.gov.companieshouse.api.accounts.model.rest.profitloss.ProfitLoss;
+import uk.gov.companieshouse.api.accounts.model.validation.Errors;
+import uk.gov.companieshouse.api.accounts.service.impl.ProfitLossService;
+import uk.gov.companieshouse.api.accounts.service.response.ResponseObject;
+import uk.gov.companieshouse.api.accounts.utility.ApiResponseMapper;
+import uk.gov.companieshouse.api.accounts.utility.ErrorMapper;
+import uk.gov.companieshouse.api.accounts.utility.LoggingHelper;
+import uk.gov.companieshouse.api.model.transaction.Transaction;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.validation.Valid;
+
+@RestController
+@RequestMapping(value = "/transactions/{transactionId}/company-accounts/{companyAccountId}/small-full/notes/profit-and-loss", produces = MediaType.APPLICATION_JSON_VALUE)
```

Comment:
This isn't a note - it's a child of current or previous period. We could achieve this functionality using a single controller by using a `RequestMapping` value of:
```
/transactions/{transactionId}/company-accounts/{companyAccountId}/small-full/{period:(current-period|previous-period)}/profit-and-loss
```

The `period` variable would be a string which could be used for logic, particularly in the `PUT` mapped method as explained below. We could even capture `period` in an enum for readability; see [here](https://www.devglan.com/spring-boot/enums-as-request-parameters-in-spring-boot-rest) for one example on how to achieve this